### PR TITLE
Mepts 273 update visit type tranformations

### PR DESCRIPTION
**Pull request by kaweesi**

### CODE REVIEW

**Comment by kaweesi** [comment]
unfortunately this lists more commits than it should maybe because i started off this work off a wrong branch that got merged into develop, its merged content however is not listed in the changes https://github.com/esaude/openmrs-database-migrator/pull/43/files

File: src/main/resources/pdiresources/visit-type/logging.ktr [hunk 1]
```diff
@@ -492,18 +497,391 @@
...
+      <enabled>Y</enabled>
+    </hop>
+    <hop>
+      <from>Filter rows</from>
+      <to>Dummy (do nothing)</to>
+      <enabled>Y</enabled>
+    </hop>
+    <hop>
+      <from>Join rows (cartesian product)</from>
+      <to>form-log</to>
```
Comment (cfaife) [inline src/main/resources/pdiresources/visit-type/logging.ktr]: Here I  think  it should be `visit-log`

File: src/main/resources/pdiresources/visit-type/logging.ktr [hunk 2]
```diff
@@ -612,8 +1004,57 @@ AND v.updated = 0;</sql>
...
     <GUI>
-      <xloc>704</xloc>
-      <yloc>128</yloc>
+      <xloc>208</xloc>
+      <yloc>112</yloc>
+      <draw>Y</draw>
+    </GUI>
+  </step>
+  <step>
+    <name>updated forms</name>
```
Comment (cfaife) [inline src/main/resources/pdiresources/visit-type/logging.ktr]: the same  here .... I  think it should be `updated visits`

File: src/main/resources/pdiresources/visit-type/preparation.ktr [hunk 3]
```diff
@@ -491,9 +491,229 @@
...
+    <partitioning>
+      <method>none</method>
+      <schema_name/>
+    </partitioning>
+    <connection>source</connection>
+    <execute_each_row>N</execute_each_row>
+    <single_statement>N</single_statement>
+    <replace_variables>N</replace_variables>
+    <quoteString>N</quoteString>
+    <sql>ALTER TABLE visit ADD column updated INT(1) DEFAULT 0;</sql>
```
Comment (cfaife) [inline src/main/resources/pdiresources/visit-type/preparation.ktr:697]: I think it should be placed `updated BOOLEAN DEFAULT 0`, to  avoid other integer values

File: src/main/resources/pdiresources/visit-type/preparation.ktr [hunk 4]
```diff
@@ -491,9 +491,229 @@
     </attributes>
   </connection>
   <order>
+    <hop>
```
Comment (cfaife) [inline src/main/resources/pdiresources/visit-type/preparation.ktr:498]: here in the data base connection we should add `set foreign_key_checks=0`; to allow delete statement

**Review by cfaife — CHANGES_REQUESTED** [review]

**Review by cfaife — APPROVED** [review]

**Pull request merged**